### ISSUE TITLE
Atoti Installation for JupyterHub

### ISSUE THREAD

**Issue by sumitrathi95** [atoti/atoti#282]
Hi Team,

Recently, I have installed atoti for Jupyterlab on my server with the help of below blog, this installation is for single user:
https://medium.com/atoti/the-ultimate-deployment-guide-for-atoti-on-aws-with-amazon-ec2-c388f0209242

I want the atoti with Jupyterlab for multiple user. So, I have installed JupyterHub on server and now able to have multiple user session but when I install and use Atoti on server and try to get session.url I am getting it as localhost.

`session.url`
`http://localhost:37337`

Please suggest me how to solve this on JupyterHub for multi user, to get URL as IP/Jupyter Server Proxy.

**Comment by tibdex** [comment]
Hello, you can use the [**url_pattern** (an optionally **inherit_global_config**) config options](https://docs.atoti.io/latest/lib/atoti.config.html#atoti.config.create_config) to change the value returned by `Session.url`.

We intend to simplify this in the next release.

**Comment by rcolle** [comment]
Hi @sumitrathi95 ,

Thanks for opening the issue!

As @tibdex mentioned, we're working on a simpler solution than setting the URL pattern for the next version of atoti.
Would you mind sharing a bit more about your setup to make sure that the proposed solution will also work in your case? In particular:
- what kind of jupyterhub deployment are you using? (littlest jupyterhub, zero to jupyterhub, something else, ...)
- is there some kind of authentication set up for your users, or can anyone freely access any user's work?
- would you like the session URL to rely on jupyter server proxy, or do you use another mechanism to expose your ports to the outside world?
- will a user share the session URL with another user (or even an external person), or do you want to limit it to the user that created the session?

Thanks!

**Comment by sumitrathi95** [comment]
Hi @tibdex and @rcolle  Thanks for quick response.
@rcolle :
- what kind of jupyterhub deployment are you using? (littlest jupyterhub, zero to jupyterhub, something else, ...)
`I am trying for littlest jupyterhub`

- is there some kind of authentication set up for your users, or can anyone freely access any user's work?
`I am using the basic authentication which comes with JupyterHub`

- would you like the session URL to rely on jupyter server proxy, or do you use another mechanism to expose your ports to the outside world?
`For single user I tried the one shared in blog (https://medium.com/atoti/the-ultimate-deployment-guide-for-atoti-on-aws-with-amazon-ec2-c388f0209242) . I think it is based on jupyter server proxy only.`

- will a user share the session URL with another user (or even an external person), or do you want to limit it to the user that created the session?
`I want to create a limit to user who created session but on server I will install atoti only once which should reflect to all users.`

**Comment by sumitrathi95** [comment]
Hello,

Can anyone please guide me how to use atoti as well as graphviz on google colab?

**Comment by sumitrathi95** [comment]
Hello Team,

I am trying to use atoti on Google Colab and able to visualize graph. I found this issue in atoti about how to use atoti with Google Colab:
https://github.com/atoti/atoti/issues/273

But when I try to use cube.schema command it is showing  below error:

`Missing Graphviz library which is required to display the graph. It can be installed with Conda: `conda install graphviz` or by following the download instructions at https://www.graphviz.org/download/.`

![image](https://user-images.githubusercontent.com/58877289/120597071-32059d80-c462-11eb-9373-3ae839231992.png)

I have installed graphviz. pydot, pygraphviz in colab still its failing.
